activerecord/test/cases/query_cache_test.rb

```diff
@@ -323,17 +322,16 @@ def test_query_cache_across_threads
 
             ActiveRecord::Base.connection_handler.clear_active_connections!(:all)
           }.call({})
+
+          assert_cache :off, thread_2_connection
         }
 
         started.wait
 
         thread_1_connection = ActiveRecord::Base.lease_connection
         assert_not_equal thread_1_connection, thread_2_connection
-        assert_cache :dirty, thread_2_connection
         checked.set
         thread.join
-
-        assert_cache :off, thread_2_connection
       }.call({})
 
```

Comment:
Since the new behavior only applies to pinned connections, we can actually restore this test to what it was and it should still pass.

Though I guess this is also fine.